builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -421,9 +421,16 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 							Default:  "scsi",
 							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
 								value := v.(string)
-								if value != "scsi" && value != "ide" {
```

Comment:
Good point.  I added a []string with each controller type and the validate func now loops over it.  I wish golang had the python-ism of "if x in y" instead of having to loop through.  Apparently that is a go-ism. See what you think about this patch.